scarb/src/resolver/mod.rs

```diff
@@ -265,7 +283,10 @@ mod tests {
             })
             .collect_vec();
 
-        let lockfile = Lockfile::new(vec![].into_iter());
+        let lockfile = Lockfile {
+            version: Default::default(),
+            packages: BTreeSet::from_iter(locks.iter().cloned()),
```

Comment:
```suggestion
            packages: locks.iter().cloned().collect(),
```

btw why can't you use `Lockfile::new` here anymore?